node/cn/backend.go

```diff
@@ -265,6 +271,21 @@ func New(ctx *node.ServiceContext, config *Config) (*CN, error) {
 	}
 	bc.SetCanonicalBlock(config.StartBlockNumber)
 
+	// Write the live pruning flag to database if the node is started for the first time
+	if config.LivePruning && !chainDB.ReadPruningEnabled() {
+		if bc.CurrentBlock().NumberU64() > 0 {
+			return nil, errors.New("cannot enable live pruning after chain has advanced")
+		}
+		chainDB.WritePruningEnabled()
+		logger.Info("Enabling live pruning")
+	}
+	// Live pruning is enabled according to the flag in database
+	// regardless of the command line flag --state.live-pruning
+	// But live pruning is disabled when --state.live-pruning-retention 0
+	if chainDB.ReadPruningEnabled() && config.LivePruningRetention != 0 {
+		logger.Info("Live pruning is enabled")
+	}
```

Comment:
@blukat29 We could consider logging when live pruning is disabled because of `--state.live-pruning-retention` set to 0 although `--state.live-pruning` was set?

Also, two consecutive logs with almost the same meaning could be confusing, although their internal semantic differs.
```
Enabling live pruning
Live pruning is enabled
```

These are not major concerns (all log-related), so I'm approving this PR. Thanks

Reply:
I will refine in another PR
```
Writing live pruning flag to database
Live pruning is enabled
Live pruning is disabled (no flag in database)
Live pruning is disbaled (retention set to zero)
```